### DISCUSSION TITLE
Managed Identity Authentication Support

### DISCUSSION THREAD

**Discussion by danbasszeti** [IEvangelist/azure-cosmos-dotnet-repository#462] [Ideas]
Hi,

I think it'd be useful if I could use Managed Identities on Azure to authenticate rather than connection strings - I'm not sure if this is already supported but I can't find it documented anywhere. It'd allow me to upgrade our security profile in accordance with a Zero-Trust policy, key couldn't be stolen etc. 

I'd be happy to help write the feature if pointed in the right direction.

**Comment by IEvangelist** [comment]
I agree, that would be a really nice addition to this library. What configuration APIs would you expect need to be updated?

**Comment by MarioMartinPlaza** [comment]
Hi @danbasszeti 

I implemented in this way:

```
services.AddCosmosRepository(options =>
 {
     if (useConnectionString)
         options.CosmosConnectionString = "ConnectionString.......";
     else
     {
         options.TokenCredential = new DefaultAzureCredential();
         options.AccountEndpoint = new Uri("https://xxxxx.documents.azure.com:443/");;
     }
     options.DatabaseId = "DatabaseId";
     options.ContainerPerItemType = true;
     options.SerializationOptions = new RepositorySerializationOptions
     {
         PropertyNamingPolicy = CosmosPropertyNamingPolicy.CamelCase
     };
 });
```

I hope this can help you
